assets/src/edit-story/components/panels/design/preset/presetPanel.js

```diff
@@ -82,11 +93,30 @@ function PresetPanel({
         },
       });
       // If no styles left, exit edit mode.
-      if (updatedStyles.length === 0) {
+      if (
+        updatedStyles.length === 0 &&
+        ((isColor && !hasLocalPresets) || isStyle)
+      ) {
         setIsEditMode(false);
       }
     },
-    [colors, isColor, textStyles, updateStory]
+    [colors, isColor, textStyles, updateStory, hasLocalPresets, isStyle]
+  );
+
+  const handleDeleteLocalPreset = useCallback(
+    (toDelete) => {
+      const updatedColors = localColors.filter((color) => color !== toDelete);
+      updateStory({
+        properties: {
+          storyPresets: { colors: updatedColors },
+        },
+      });
+      // If no colors are left, exit edit mode.
+      if (updatedColors.length === 0 && globalPresets.length === 0) {
+        setIsEditMode(false);
+      }
+    },
+    [globalPresets.length, localColors, updateStory]
   );
```

Comment:
I think these two should go in an external hook as well - just for the overview.